### PR TITLE
[WIP] convert to typescript

### PR DESCRIPTION
**Pull request by VinSpee**
I noticed a few things while working on `Zenith`:

1. crushinator helpers obviously doesnt ship with type definitions
2. it's nice when projects ship with type definitions so my editor tells
   me the write/wrong things to do

This commit converts the project to typescript, and uses tsdx (https://github.com/jaredpalmer/tsdx) as the build tooling, which is a light wrapper around a parcel config.

I _figure_ this will be a major version release for safety's sake, so
I removed the _long_ deprecated `string` configuration.

prior to release, I'd like to get [typedoc](https://github.com/tom-grey/typedoc-plugin-markdown) working too

### CODE REVIEW

**Comment by VinSpee** [comment]
thoughts @redoPop @jonbellah?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 /node_modules
 /coverage
 /spec
+/junit.xml
+/dist
```
Comment (redoPop) [inline .gitignore]: Since we're now excluding distribution files from the repo, we should remove `bower.json` 😆

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 /node_modules
 /coverage
 /spec
+/junit.xml
```
Comment (redoPop) [inline .gitignore]: This file shouldn't exist according to the jest-junit config in package.json. Is something up?

**Comment by jonbellah** [comment]
:bigfan:

I've been using this lib a bit on a TypeScript project over the last few weeks, actually, and have found the lack of types to be a pretty big pain point. It's one of the only libs I consistently find myself having to triple check the docs on because I don't have that extra type safety net (or the autocomplete/options list that you mentioned).

I'll avoid wading into the discussion about maintenance load, since I haven't traditionally been a maintainer of this lib.

**Comment by VinSpee** [comment]
@redopop I was anticipating that reaction and while I’m less worried about that I think it is a fair point. 

I think including a minimal, hand-authored version of the `index.d.ts` would suffice to cover the ~~pain~~ moderate inconvenience. Would you be open to that kind of PR Joe?

**Comment by redoPop** [comment]
That sounds like a good compromise to me. It's rare that we change the footprint of these public helper methods so the manual maintenance work for the definitions should be low.

**Pull request closed without merging**